frontend/src/metabase/dashboard/components/DashboardHeader/DashboardHeaderButtonRow/action-buttons.tsx

```diff
@@ -0,0 +1,224 @@
+import { withRouter } from "react-router";
+
+import { RefreshWidgetButton } from "../../DashboardActions.styled";
+import { DashboardBookmark } from "../../DashboardBookmark";
+import { DashboardEmbedAction } from "../../DashboardEmbedAction";
+import { ExtraEditButtonsMenu } from "../../ExtraEditButtonsMenu";
+import { DashboardHeaderActionDivider } from "../DashboardHeader.styled";
+import {
+  AddActionElementButton,
+  AddFilterParameterButton,
+  AddHeadingOrTextButton,
+  AddLinkCardButton,
+  AddQuestionButton,
+  AddSectionButton,
+  AddTemporalUnitButton,
+  CopyAnalyticsDashboardButton,
+  DashboardActionMenu,
+  DashboardInfoButton,
+  DashboardSubscriptionButton,
+  EditDashboardButton,
+  FullscreenAnalyticsDashboard,
+  FullscreenToggle,
+  getExtraButtons,
+  NightModeToggleButton,
+  shouldRenderSubscriptionButton,
+} from "../buttons";
+
+import type {
+  DashboardActionButton,
+  DashboardActionKey,
+  HeaderButtonProps,
+} from "./types";
+
+export const DASHBOARD_ACTION = {
+  ADD_QUESTION: "ADD_QUESTION",
+  ADD_HEADING_OR_TEXT: "ADD_HEADING_OR_TEXT",
+  ADD_LINK_CARD: "ADD_LINK_CARD",
+  ADD_SECTION: "ADD_SECTION",
+  ADD_TEMPORAL_UNIT: "ADD_TEMPORAL_UNIT",
+  ADD_FILTER_PARAMETER: "ADD_FILTER_PARAMETER",
+  ADD_ACTION_ELEMENT: "ADD_ACTION_ELEMENT",
+  EXTRA_EDIT_BUTTONS_MENU: "EXTRA_EDIT_BUTTONS_MENU",
+  COPY_ANALYTICS_DASHBOARD: "COPY_ANALYTICS_DASHBOARD",
+  EDIT_DASHBOARD: "EDIT_DASHBOARD",
+  DASHBOARD_SUBSCRIPTION: "DASHBOARD_SUBSCRIPTION",
+  DASHBOARD_EMBED_ACTION: "DASHBOARD_EMBED_ACTION",
+  REFRESH_WIDGET: "REFRESH_WIDGET",
+  NIGHT_MODE_TOGGLE: "NIGHT_MODE_TOGGLE",
+  FULLSCREEN_TOGGLE: "FULLSCREEN_TOGGLE",
+  DASHBOARD_HEADER_ACTION_DIVIDER: "DASHBOARD_HEADER_ACTION_DIVIDER",
+  DASHBOARD_BOOKMARK: "DASHBOARD_BOOKMARK",
+  DASHBOARD_INFO: "DASHBOARD_INFO",
+  DASHBOARD_ACTION_MENU: "DASHBOARD_ACTION_MENU",
+  FULLSCREEN_ANALYTICS_DASHBOARD: "FULLSCREEN_ANALYTICS_DASHBOARD",
+} as const;
+
+export const dashboardActionButtons: Record<
+  DashboardActionKey,
+  DashboardActionButton
+> = {
+  // ACTIONS WHEN EDITING DASHBOARD
+  [DASHBOARD_ACTION.ADD_QUESTION]: {
+    component: AddQuestionButton,
+    enabled: ({ isEditing }) => isEditing,
```

Comment:
nit: This actually repeats the logic we did in https://github.com/metabase/metabase/blob/58a338d42ae3bdb5bbe18e378b3a5021390ffd43/frontend/src/metabase/dashboard/components/DashboardHeader/DashboardHeaderButtonRow/DashboardHeaderButtonRow.tsx#L41-L43

I'm on the fence about this, so there's no definitive answer from me.

Either leave this as is or remove all `isEditing` conditions.

e.g.
```suggestion
    enabled: () => true,
```

Reply:
Yeah I've been thinking about this quite a lot as well. Basically the two sides for me are:

*   Keep: in case they are needed when customizing the SDK (i.e. maybe we show the button, but disabled?). What I mean to say is that there's a case that these buttons can be independent of each other and I want to make sure that it's available
*   Nuke: we already have the `isEditing` condition so we don't have to have it for each button. But of course that takes away "button independence" but is it something we need?

Reply:
you have a point. Let's just keep it as is then.